tools/doc-generator/README.md

```diff
@@ -0,0 +1,23 @@
+# Documentation generation tool
+
+The doc-generator tool is used to automatically generate the configuration flags documentation from the metadata information
+provided in the code. 
+
+## Run 
+
+The tool receives as input a template file and generates the list of configuration value blocks: 
+
+```shell
+go run ./tools/doc-generator docs/sources/configuration/index.template > docs/sources/configuration/_index.md
```

Comment:
looks great ;) however, we could simplify adding a target to Makefile that would do everything that we need ))

Reply:
There is already a Makefile target: https://github.com/grafana/loki/blob/main/Makefile#L732-L733